lib/OpenQA/Schema/ResultSet/Comments.pm

```diff
@@ -41,4 +41,43 @@ sub referenced_bugs {
     return \%bugrefs;
 }
 
+=over 4
+
+=item comment_data_for_jobs($jobs)
+
+Return a hashref with bugrefs, labels and the number of regular comments per job ID.
+
+=back
+
+=cut
+
+sub comment_data_for_jobs ($self, $jobs) {
+    my @job_ids  = map { $_->id } ref $jobs eq 'ARRAY' ? @$jobs : $jobs->all;
+    my $comments = $self->search({job_id => {in => \@job_ids}}, {order_by => 'me.id'});
+    my $bugs     = $self->result_source->schema->resultset('Bugs');
+
+    my (%res, %bugdetails);
+    while (my $comment = $comments->next) {
+        my ($bugrefs, $res) = ($comment->bugrefs, $res{$comment->job_id} //= {});
+        if (@$bugrefs) {
+            my $bugs_of_job = ($res->{bugs} //= {});
+            for my $bug (@$bugrefs) {
+                $bugdetails{$bug} = $bugs->get_bug($bug) unless exists $bugdetails{$bug};
+                $bugs_of_job->{$bug} = 1;
+            }
+            $res->{bugdetails} = \%bugdetails;
+            $res->{reviewed}   = 1;
+        }
+        elsif (my $label = $comment->label) {
+            $res->{label}    = $label;
+            $res->{reviewed} = 1;
+        }
+        else {
+            $res->{comments}++;
+        }
+        # note: Previous labels are overwritten here so only the most recent label is returned.
+    }
+    return \%res;
```

Comment:
maybe a stupid question,  I don't find the place which sets  values for `%res` in this function.

Reply:
For example line 76.

$foo->{bar} sets the value for the key "bar" in a hash %foo

Reply:
I thoguht `$res->{comments}++`  was used to set the value for the `$res` in line 61 `my ($bugrefs, $res) = ($comment->bugrefs, $res{$comment->job_id} //= {});`

Reply:
Ok, sorry, I think I did not look or explain well enough You took a much better look than I did :)

So

```perl
my ($bugrefs, $res) = ($comment->bugrefs, $res{$comment->job_id} //= {});
```

adds a reference `$res` to `$res{$comment->job_id}` and AFAICS at the same time it initializes `$res{$comment->job_id}` to an empty hash if it has no value yet.

Over the reference `$res` pointing to individual values, the complete hash `%res` is updated in lines like `$res->{comments}++`.

Good enough? :)

Reply:
Thanks for your explanation. It's really good.